File: lib/models/category.dart
enum CategoryType { income, expense }

class Category {
  final String id;
  final String name;
  final String iconPath;
  final CategoryType type;
  final DateTime createdAt;
  final DateTime updatedAt;

  Category({
    required this.id,
    required this.name,
    required this.iconPath,
    required this.type,
    required this.createdAt,
    required this.updatedAt,
  });
}

File: backend/src/main/java/com/walletapp/demo/entity/Category.java
package com.walletapp.demo.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import com.walletapp.demo.enums.CategoryType;

@Entity
@Table(name = "categories", indexes = {
        @Index(name = "idx_category_type", columnList = "category_type"),

})
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Category {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "category_id")
    private Long id;

    private String name;

    @Enumerated(EnumType.STRING)
    @Column(name = "category_type", nullable = false)
    private CategoryType type;

    @Column(name = "icon_key", nullable = false)
    private String iconKey;

    @CreationTimestamp
    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
